diff -Naur gmp-tarballs/gmp-6.1.2/configure gmpbuild/configure
--- gmp-tarballs/gmp-6.1.2/configure	2016-12-16 16:45:32.000000000 +0100
+++ gmpbuild/configure	2020-02-07 12:55:27.853718384 +0100
@@ -4087,8 +4087,8 @@
 #
 cclist="gcc cc"
 
-gcc_cflags="-O2 -pedantic"
-gcc_64_cflags="-O2 -pedantic"
+gcc_cflags="-O2 -pedantic -fPIC"
+gcc_64_cflags="-O2 -pedantic -fPIC"
 cc_cflags="-O"
 cc_64_cflags="-O"
 
@@ -27270,6 +27270,9 @@
 
           $as_echo "#define HAVE_HOST_CPU_FAMILY_x86_64 1" >>confdefs.h
 
+echo "define(<PIC_ALWAYS>,<yes>)" >> $gmp_tmpconfigm4
+
+
 	  case $host in
 	    *-*-darwin*)
 
@@ -28181,7 +28184,7 @@
 # FIXME: Upcoming version of autoconf/automake may not like broken lines.
 #        Right now automake isn't accepting the new AC_CONFIG_FILES scheme.
 
-ac_config_files="$ac_config_files Makefile mpf/Makefile mpn/Makefile mpq/Makefile mpz/Makefile printf/Makefile scanf/Makefile rand/Makefile cxx/Makefile tests/Makefile tests/devel/Makefile tests/mpf/Makefile tests/mpn/Makefile tests/mpq/Makefile tests/mpz/Makefile tests/rand/Makefile tests/misc/Makefile tests/cxx/Makefile doc/Makefile tune/Makefile demos/Makefile demos/calc/Makefile demos/expr/Makefile gmp.h:gmp-h.in"
+ac_config_files="$ac_config_files Makefile mpf/Makefile mpn/Makefile mpq/Makefile mpz/Makefile printf/Makefile scanf/Makefile rand/Makefile cxx/Makefile tests/Makefile tests/devel/Makefile tests/mpf/Makefile tests/mpn/Makefile tests/mpq/Makefile tests/mpz/Makefile tests/rand/Makefile tests/misc/Makefile tests/cxx/Makefile tune/Makefile demos/Makefile demos/calc/Makefile demos/expr/Makefile gmp.h:gmp-h.in"
 
 cat >confcache <<\_ACEOF
 # This file is a shell script that caches the results of configure
@@ -29325,7 +29328,6 @@
     "tests/rand/Makefile") CONFIG_FILES="$CONFIG_FILES tests/rand/Makefile" ;;
     "tests/misc/Makefile") CONFIG_FILES="$CONFIG_FILES tests/misc/Makefile" ;;
     "tests/cxx/Makefile") CONFIG_FILES="$CONFIG_FILES tests/cxx/Makefile" ;;
-    "doc/Makefile") CONFIG_FILES="$CONFIG_FILES doc/Makefile" ;;
     "tune/Makefile") CONFIG_FILES="$CONFIG_FILES tune/Makefile" ;;
     "demos/Makefile") CONFIG_FILES="$CONFIG_FILES demos/Makefile" ;;
     "demos/calc/Makefile") CONFIG_FILES="$CONFIG_FILES demos/calc/Makefile" ;;
diff -Naur gmp-tarballs/gmp-6.1.2/configure.ac gmpbuild/configure.ac
--- gmp-tarballs/gmp-6.1.2/configure.ac	2016-12-16 16:45:27.000000000 +0100
+++ gmpbuild/configure.ac	2020-02-07 12:55:27.853718384 +0100
@@ -3696,6 +3696,7 @@
         64|x32)
           GMP_INCLUDE_MPN(x86_64/x86_64-defs.m4)
           AC_DEFINE(HAVE_HOST_CPU_FAMILY_x86_64)
+	  GMP_DEFINE_RAW(["define(<PIC_ALWAYS>,<yes>)"])
 	  case $host in
 	    *-*-darwin*)
 	      GMP_INCLUDE_MPN(x86_64/darwin.m4) ;;
diff -Naur gmp-tarballs/gmp-6.1.2/Makefile.am gmpbuild/Makefile.am
--- gmp-tarballs/gmp-6.1.2/Makefile.am	2016-12-16 16:45:27.000000000 +0100
+++ gmpbuild/Makefile.am	2020-02-07 12:55:27.853718384 +0100
@@ -110,7 +110,7 @@
 LIBGMPXX_LT_AGE      = 5
 
 
-SUBDIRS = tests mpn mpz mpq mpf printf scanf rand cxx demos tune doc
+SUBDIRS = tests mpn mpz mpq mpf printf scanf rand cxx demos tune
 
 EXTRA_DIST = configfsf.guess configfsf.sub .gdbinit INSTALL.autoconf \
 	     COPYING.LESSERv3 COPYINGv2 COPYINGv3
diff -Naur gmp-tarballs/gmp-6.1.2/Makefile.in gmpbuild/Makefile.in
--- gmp-tarballs/gmp-6.1.2/Makefile.in	2016-12-16 16:45:34.000000000 +0100
+++ gmpbuild/Makefile.in	2020-02-07 12:55:27.853718384 +0100
@@ -566,7 +566,7 @@
 LIBGMPXX_LT_CURRENT = 9
 LIBGMPXX_LT_REVISION = 2
 LIBGMPXX_LT_AGE = 5
-SUBDIRS = tests mpn mpz mpq mpf printf scanf rand cxx demos tune doc
+SUBDIRS = tests mpn mpz mpq mpf printf scanf rand cxx demos tune
 
 # The "test -f" support for srcdir!=builddir is similar to the automake .c.o
 # etc rules, but with each foo.c explicitly, since $< is not portable
diff -Naur gmp-tarballs/gmp-6.1.2/mpn/asm-defs.m4 gmpbuild/mpn/asm-defs.m4
--- gmp-tarballs/gmp-6.1.2/mpn/asm-defs.m4	2016-12-16 16:45:27.000000000 +0100
+++ gmpbuild/mpn/asm-defs.m4	2020-02-07 12:55:27.853718384 +0100
@@ -1051,7 +1051,7 @@
 dnl  systems which are always PIC.  PIC_ALWAYS established in config.m4
 dnl  identifies these for us.
 
-ifelse(`PIC_ALWAYS',`yes',`define(`PIC')')
+ifelse(PIC_ALWAYS,yes,`define(`PIC')')
 
 
 dnl  Various possible defines passed from the Makefile that are to be tested
